app/controllers/checkouts_controller.rb

```diff
@@ -3,14 +3,18 @@ class CheckoutsController < ApplicationController
 
   def create
     @book = Book.find(params[:book_id])
-    checkout = Checkout.create(book: @book, user: current_user)
+    CheckOutBook.process(book: @book, user: current_user)
+    redirect_to root_path, success: t('.successful_message')
+  end
 
-    redirect_to root_path
+  def update
```

Comment:
I would probably put this in a separate resource like `:renewals` and do it on `create` so you're "creating a renewal" , rather than "updating a checkout" (which is actually renewing it)